supportedBackends/bcs/src/main/scala/cromwell/backend/impl/bcs/BcsConfiguration.scala

```diff
@@ -1,37 +1,110 @@
 package cromwell.backend.impl.bcs
 
-import com.aliyuncs.batchcompute.main.v20151111.{BatchComputeClient}
+import com.aliyuncs.auth.BasicSessionCredentials
+import com.aliyuncs.auth.BasicCredentials
+
+import com.aliyuncs.batchcompute.main.v20151111.BatchComputeClient
+import com.typesafe.config.{Config, ConfigFactory}
 import cromwell.backend.BackendConfigurationDescriptor
 import net.ceedubs.ficus.Ficus._
-
+import cromwell.engine.backend.BackendConfigurationEntry
+import cromwell.backend.impl.bcs.callcaching.{CopyCachedOutputs, UseOriginalCachedOutputs}
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+import cromwell.core.DockerConfiguration
 
 object BcsConfiguration{
   val OssEndpointKey = "ossEndpoint"
   val OssIdKey = "ossId"
   val OssSecretKey = "ossSecret"
   val OssTokenKey = "ossToken"
+  val clientExpireTime: Long = 30 * 60
+  def currentTimestamp = System.currentTimeMillis / 1000
+
+  private def BackendConfig = {
+    ConfigFactory.invalidateCaches
+    ConfigFactory.load.getConfig("backend")
+  }
+
+  private def BackendProviders = BackendConfig.getConfig("providers")
+
+  private def BackendNames: Set[String] = BackendProviders.entrySet().asScala.map(_.getKey.split("\\.").toSeq.head).toSet
+
+  def AllBackendEntries: List[BackendConfigurationEntry] = BackendNames.toList map { backendName =>
+    val entry = BackendProviders.getConfig(backendName)
+    BackendConfigurationEntry(
+      backendName,
+      entry.getString("actor-factory"),
+      entry.as[Option[Config]]("config").getOrElse(ConfigFactory.empty("empty"))
+    )
+  }
+
+  def backendConfigurationDescriptor(backendName: String): Try[BackendConfigurationDescriptor] = {
+    AllBackendEntries.collect({case entry if entry.name.equalsIgnoreCase(backendName) => entry.asBackendConfigurationDescriptor}).headOption match {
+      case Some(descriptor) => Success(descriptor)
+      case None => Failure(new Exception(s"invalid backend: $backendName"))
+    }
+  }
+
+  def bcsConfig: Option[Config] = BcsConfiguration.backendConfigurationDescriptor("BCS") match {
+    case Success(conf) => Some(conf.backendConfig)
+    case Failure(_) => None
+  }
 }
 
 final class BcsConfiguration(val configurationDescriptor: BackendConfigurationDescriptor) {
+
   val runtimeConfig = configurationDescriptor.backendRuntimeAttributesConfig
+
   val bcsRegion: Option[String] = configurationDescriptor.backendConfig.as[Option[String]]("region")
 
   val bcsUserDefinedRegion: Option[String] = configurationDescriptor.backendConfig.as[Option[String]]("user-defined-region")
 
   val bcsUserDefinedDomain: Option[String] = configurationDescriptor.backendConfig.as[Option[String]]("user-defined-domain")
 
-  val bcsAccessId: Option[String] = configurationDescriptor.backendConfig.as[Option[String]]("access-id")
+  def bcsAccessId: Option[String] = BcsConfiguration.bcsConfig flatMap {c => c.as[Option[String]]("access-id")}
```

Comment:
Why didn't the previous implementation work? This appears read from ONLY a backend config named `BCS` instead of from the passed in configuration descriptor.

Reply:
> Why didn't the previous implementation work? This appears read from ONLY a backend config named `BCS` instead of from the passed in configuration descriptor.

The new implementation is aimed to refresh credentials by reading the latest config. A more graceful solution will be provided later on.